src/main/python/github_scraper.py

```diff
@@ -1,28 +1,47 @@
-from github import Github
+import os
+import requests
+from collections import namedtuple
+
 GITHUB_AUTH_TOKEN = 'dummy'
-github_client = Github(GITHUB_AUTH_TOKEN)
+GithubItem = namedtuple('GithubItem', 'file_name file_url repository_name repository_owner')
+
+
+def compute_directory_name(phrase):
+    terms = phrase.split()
+    return '_'. join(terms)
+
 
+def create_directory_for_search_phrase(directory_name):
+    full_directory_path = f'../resources/{directory_name}'
+    if not os.path.exists(full_directory_path):
+        os.makedirs(full_directory_path)
 
-def search_github(keyword):
-    rate_limit = github_client.get_rate_limit()
-    rate = rate_limit.search
-    if rate.remaining == 0:
-        print(f'You have 0/{rate.limit} API calls remaining. Reset time: {rate.reset}')
-        return
-    else:
-        print(f'You have {rate.remaining}/{rate.limit} API calls remaining')
 
-    query = f'"{keyword} english" in:file extension:po'
-    result = github_client.search_code(query, order='desc')
+def download_files(github_files, search_phrase):
+    directory_name = compute_directory_name(search_phrase)
+    create_directory_for_search_phrase(directory_name)
+    for github_file in github_files:
+        url = github_file.file_url
+        headers = {'accept': 'application/vnd.github.VERSION.raw'}
+        r = requests.get(url, headers=headers)
+        open(f'../resources/{directory_name}/{github_file.file_name}', 'wb').write(r.content)
```

Comment:
When you use with statement with open function, you do not need to close the file at the end, because with would automatically close it for you.

Example:
with open(f'../resources/{directory_name}/{github_file.file_name}', 'wb') as write_file:
     write_file.write(r.content)

Reply:
Looks like that's not case in this piece of code, but is on other